### PR TITLE
PB-5664: Fix concurrent map writes issue in backup tests

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
I saw that the ginkgo failed with the below error in the IBM pipeline: 
```
21:59:25  2024-02-14 16:29:24 +0000:[INFO] [{MultipleInPlaceRestoreSameTime}] Issuing multiple in-place restore at the same time
21:59:25  fatal error: concurrent map writes
21:59:25  fatal error: concurrent map writes
21:59:25  goroutine 11366 [running]:
21:59:25  github.com/portworx/torpedo/tests/backup.glob..func68.2.5.1({0xc0023185a0, 0x25}, {0xc00251f3c0, 0x35})
21:59:25  	/go/src/github.com/portworx/torpedo/tests/backup/backup_restore_basic_test.go:1963 +0x312
21:59:25  created by github.com/portworx/torpedo/tests/backup.glob..func68.2.5 in goroutine 11070
21:59:25  	/go/src/github.com/portworx/torpedo/tests/backup/backup_restore_basic_test.go:1959 +0x13a
```

- [x] Fix concurrent map writes issue in backup tests

Dashboard URL:
https://aetos.pwx.purestorage.com/resultSet/testSetID/546628  (All passed other than)
https://aetos.pwx.purestorage.com/resultSet/testSetID/547618 (Only MultipleInPlaceRestoreSameTime)
https://aetos.pwx.purestorage.com/resultSet/testSetID/547640 (All testcase run with  final changes)

**Which issue(s) this PR fixes** (optional)
Closes #PB-5664, PB-5321

**Special notes for your reviewer**:

### CODE REVIEW

**Review by atrivedi-px — APPROVED** [review]
LGTM, We may need to make changes to backup_helper in future to handle this from CreateRestoreWithValidation itself

**Comment by kshithijiyer-px** [comment]
> LGTM, We may need to make changes to backup_helper in future to handle this from CreateRestoreWithValidation itself

Yes when we refactor `CreateRestoreWithValidation()` we'll need to update it.

**Review by snigdha-px — APPROVED** [review]
LGTM

File: tests/backup/backup_restore_basic_test.go [hunk 1]
```diff
@@ -1932,18 +1933,21 @@ var _ = Describe("{MultipleInPlaceRestoreSameTime}", func() {
...
 			for _, bkpNameSpace := range bkpNamespaces {
+				time.Sleep(timeBetweenConsecutiveOps)
 				sem <- struct{}{}
-				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, bkpNameSpace, RandomString(5))
 				wg.Add(1)
+				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, bkpNameSpace, RandomString(5))
 				go func(bkpNameSpace string, backupName string) {
 					defer GinkgoRecover()
 					defer wg.Done()
-					defer func() { <-sem }()
```
Comment (kphalgun-px) [inline tests/backup/backup_restore_basic_test.go:1942]: I do not think there would be any use for the variable 'sem' if we remove this.

Comment (kshithijiyer-px) [reply tests/backup/backup_restore_basic_test.go:1942]: Make sense let me add it back.

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**